src/app/share-preview/components/share-preview/share-preview.component.html

```diff
@@ -109,11 +109,18 @@
         </div>
         <button class="btn btn-alternate" *ngIf="isRelationshipShare">Log in</button>
       </strong>
-      <strong *ngIf="isLinkShare && isLoggedIn && !hasRequested">Request full access from {{shareAccount.fullName}}.</strong>
+      <strong *ngIf="isLinkShare && isLoggedIn && !hasRequested">
+        <ng-container *ngIf="sharePreviewVO?.autoApproveToggle === 1 ; else requestFullAccessText">
+          Accept and view the share in your archive.
+        </ng-container>
+        <ng-template #requestFullAccessText>
+          Request full access from {{shareAccount.fullName}}.
+        </ng-template>
+      </strong>
       <strong *ngIf="isLinkShare && isLoggedIn && hasRequested">Your request is awaiting approval from {{shareAccount.fullName}}.</strong>
       <strong *ngIf="isRelationshipShare && isLoggedIn">Your current archive does not have access to this content. Switch to view or edit.</strong>
       <div *ngIf="isLinkShare && isLoggedIn">
-        <button class="btn btn-alternate" (click)="onRequestAccessClick()" [disabled]="waiting || hasRequested" *ngIf="isLinkShare">{{ hasRequested ? 'Access requested' : 'Request access' }}</button>
+        <button class="btn btn-alternate" (click)="onRequestAccessClick()" [disabled]="waiting || hasRequested" *ngIf="isLinkShare">{{ hasRequested ? 'Access requested' : requestAccessText }}</button>
```

Comment:
I see at least four different approaches to varying text in this file:

- HTML elements with `*ngIf` expressions to show/hide
- HTML elements with `*ngIf="(expr) ; else templateName` and a partner `<ng-template>`
- strings selected via the conditional operator, eg `{{ (expr) ? 'string1' : 'string2' }}`
- strings injected via code, eg `chooseArchiveText` and `requestAccessText`

That seems like... kind of a lot. Do we need all of those different approaches to live side-by-side?

Personally, I quite like the first: `*ngIf` attributes with simple boolean expressions to either show or hide the element: it's simple, straightforward, and easy to reason about. In general, my default stance is that text belongs in the component HTML, rather than in code, and the component class should have the appropriately-named booleans to select the desired behavior.

In this specific case, of wanting a single button with various text content, can we use an [`<ng-container>`](https://angular.io/api/core/ng-container) to hang the `*ngIf` conditions on?

Or maybe I'm missing something, in which case I'd love to learn! Is there some drawback to the approach I am suggesting? Is the complexity here merited because the logic of which string to show is equally complex?